cmd/docgen/docgen.go

```diff
@@ -124,3 +128,153 @@ func runDocGenCommandFunc(clients *shared.ClientFactory, cmd *cobra.Command, arg
 	))
 	return nil
 }
+
+// genMarkdownTree creates markdown reference of commands for the docs site.
+//
+// Reference: https://github.com/spf13/cobra/blob/3f3b81882534a51628f3286e93c6842d9b2e29ea/doc/md_docs.go#L119-L158
+func genMarkdownTree(cmd *cobra.Command, fs afero.Fs, dir string) error {
+	for _, c := range cmd.Commands() {
+		if !c.IsAvailableCommand() || c.IsAdditionalHelpTopicCommand() {
+			continue
+		}
+		if err := genMarkdownTree(c, fs, dir); err != nil {
+			return err
+		}
+	}
+	basename := strings.ReplaceAll(cmd.CommandPath(), " ", "_") + ".md"
+	filename := filepath.Join(dir, basename)
+	f, err := fs.Create(filename)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	if err := genMarkdownCommand(cmd, f); err != nil {
+		return err
+	}
+	return nil
+}
+
+// genMarkdownCommand creates custom markdown output for a command.
+//
+// Reference: https://github.com/spf13/cobra/blob/3f3b81882534a51628f3286e93c6842d9b2e29ea/doc/md_docs.go#L56-L117
+func genMarkdownCommand(cmd *cobra.Command, w io.Writer) error {
+	cmd.InitDefaultHelpCmd()
+	cmd.InitDefaultHelpFlag()
+
+	buf := new(bytes.Buffer)
+	name := cmd.CommandPath()
+
+	fmt.Fprintf(buf, "# `%s`\n\n", name)
+	fmt.Fprintf(buf, "%s\n\n", cmd.Short)
+	if len(cmd.Long) > 0 {
+		fmt.Fprintf(buf, "## Description\n\n")
+		description, err := render(cmd.Long)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(buf, "%s\n\n", description)
+	}
+	if cmd.Runnable() {
+		fmt.Fprintf(buf, "```\n%s\n```\n\n", cmd.UseLine())
+	}
+	if len(cmd.Example) > 0 {
+		fmt.Fprintf(buf, "## Examples\n\n")
+		fmt.Fprintf(buf, "```\n%s\n```\n\n", cmd.Example)
+	}
+	if err := genMarkdownCommandFlags(buf, cmd); err != nil {
+		return err
+	}
+	if hasSeeAlso(cmd) {
+		fmt.Fprintf(buf, "## See also\n\n")
```

Comment:
ignore-me: While I prefer "Title Case", this looks correct because our docs use "Sentence case". I think it's best for the CLI to stay aligned with our docs. So please ignore me! 🧠